tests/model_create_test.js

```diff
@@ -0,0 +1,476 @@
+      name: { type: DataTypes.STRING, allowNull: false }
+    });
+
+    await this.sequelize.sync({ force: true });
+  });
+
+  describe('findOrCreate', () => {
+    // Reason: CockroachDB guarantees that while a transaction is pending, it is isolated from other concurrent transactions with serializable isolation.
+    // https://www.cockroachlabs.com/docs/stable/transactions.html
+    it.skip('supports transactions', async function() {
+      const t = await this.sequelize.transaction();
+
+      await this.User.findOrCreate({
+        where: {
+          username: 'Username'
+        },
+        defaults: {
+          data: 'some data'
+        },
+        transaction: t
+      });
+
+      const count = await this.User.count();
+      expect(count).to.equal(0);
+      await t.commit();
+      const count0 = await this.User.count();
+      expect(count0).to.equal(1);
+    });
+
+    // Reason: CRDB has to return a Details field at bytes stream. Since it doesn't, it does not
+    // generate the 'error.errors' array. 
+    // https://github.com/cockroachdb/cockroach/issues/63332
+    it.skip('should error correctly when defaults contain a unique key and the where clause is complex', async function() {
+      const User = this.sequelize.define('user', {
+        objectId: {
+          type: DataTypes.STRING,
+          unique: true
+        },
+        username: {
+          type: DataTypes.STRING,
+          unique: true
+        }
+      });
+
+      await User.sync({ force: true });
+      await User.create({ username: 'gottlieb' });
+
+      try {
+        await User.findOrCreate({
+          where: {
+            [Op.or]: [{
+              objectId: 'asdasdasd1'
+            }, {
+              objectId: 'asdasdasd2'
+            }]
+          },
+          defaults: {
+            username: 'gottlieb'
+          }
+        });
+      } catch (error) {
+        expect(error).to.be.instanceof(Sequelize.UniqueConstraintError);
+        expect(error.errors[0].path).to.be.a('string', 'username');
+      }
+    });
+  
+    // Reason: Results of concurrent transactions are not deterministic.
+    // Probably will have to refactor this test to work with threads or some retrying strategy.
+    it.skip('should not deadlock with concurrency duplicate entries and no outer transaction', async function() {
+      const User = this.sequelize.define('User', {
+        email: {
+          type: DataTypes.STRING,
+          unique: 'company_user_email'
+        },
+        companyId: {
+          type: DataTypes.INTEGER,
+          unique: 'company_user_email'
+        }
+      });
+
+      await User.sync({ force: true });
+
+      await Promise.all(_.range(50).map(() => {
+        return User.findOrCreate({
+          where: {
+            email: 'unique.email.1@sequelizejs.com',
+            companyId: 2
+          }
+        });
+      }));
+    });
+
+    // Reason: Transaction stuff. Usually crashes Sequelize and times out tests by 60 seconds.    
```

Comment:
I think these limitations are worth documenting in the read me.

Reply:
I think they already are. I'll try to just explain it more clearly there.